solidity/dashboard/src/pages/rewards/RewardsOverviewPage.jsx

```diff
@@ -1,42 +1,86 @@
-import BeaconRewardsDetails from "../../components/BeaconRewardsDetails"
+import { SubmitButton } from "../../components/Button"
+import {
+  BeaconRewardsDetails,
+  ECDSARewardsDetails,
+} from "../../components/RewardsDetails"
 import StakeDropChart from "../../components/StakeDropChart"
 import { useWeb3Context } from "../../components/WithWeb3Context"
 import { TokenAmountSkeleton } from "../../components/skeletons"
+import EmptyStatePage from "./EmptyStatePage"
 
 const RewardsOverviewPage = () => {
   const dispatch = useDispatch()
   const { yourAddress } = useWeb3Context()
-  const { beaconRewardsFetching, becaonRewardsBalance } = useSelector(
-    (state) => state.rewards
-  )
+  const {
+    beaconRewardsFetching,
+    becaonRewardsBalance,
+    ecdsaDistributedBalance,
+    ecdsaAvailableRewardsFetching,
+    ecdsaAvailableRewardsBalance,
+    ecdsaAvailableRewards,
+  } = useSelector((state) => state.rewards)
 
   useEffect(() => {
     dispatch({
       type: "rewards/beacon_fetch_distributed_rewards_request",
       payload: yourAddress,
     })
+    dispatch({
+      type: "rewards/ecdsa_fetch_distributed_rewards_request",
+      payload: yourAddress,
+    })
+    dispatch({
+      type: "rewards/ecdsa_fetch_available_rewards_request",
+      payload: yourAddress,
+    })
   }, [dispatch, yourAddress])
 
+  const onWithdrawECDSARewards = useCallback(
+    (awaitingPromise) => {
+      dispatch({
+        type: "rewards/ecdsa_withdraw",
+        payload: ecdsaAvailableRewards,
+        meta: awaitingPromise,
+      })
+    },
+    [dispatch, ecdsaAvailableRewards]
+  )
+
   return (
-    <section className="rewards-overview--random-beacon">
-      <section className="tile">
-        <Balance
-          title="Random Beacon Rewards"
-          rewardsBalance={becaonRewardsBalance}
-          isBalanceFetching={beaconRewardsFetching}
-        />
-        <section className="mt-2">
-          <BeaconRewardsDetails />
+    <>
+      <section className="rewards-overview--random-beacon">
+        <section className="tile">
+          <Balance
+            title="Random Beacon Rewards"
+            rewardsBalance={becaonRewardsBalance}
+            isBalanceFetching={beaconRewardsFetching}
+          />
+          <section className="mt-2">
+            <BeaconRewardsDetails />
+          </section>
+        </section>
+        <section className="tile">
+          <StakeDropChart />
         </section>
       </section>
-      <section className="tile">
-        <StakeDropChart />
+
+      <section className="rewards-overview--ecdsa">
+        <section className="tile">
+          <Balance
+            title="tBTC Rewards"
+            rewardsBalance={ecdsaAvailableRewardsBalance}
+            isBalanceFetching={ecdsaAvailableRewardsFetching}
+            onWithdrawAll={onWithdrawECDSARewards}
+          />
+        </section>
+        <section className="tile">
+          <ECDSARewardsDetails pastRewards={ecdsaDistributedBalance} />
```

Comment:
Were we intending on also displaying the rewards chart for tBTC? The distribution is different/higher here than for the beacon rewards... We should probably have neither or both (and it seems like neither is probably easier).

Net net we should have a consistent layout for the tBTC and beacon rewards, even if we don't have the same info available in both cases. I know for the beacon you dropped the middle pane because we were displaying less info, but I think the current result is strictly worse, as it's a bit visually unstable.

Reply:
so what's a decision about chart for tBTC or we are going to drop beacon chart, thoughts?

Reply:
If it's quick and easy I'd be fine with having the rewards chart for tBTC, but I also want to get this out last week, so I'm fine dropping the chart for the beacon. This is strictly a time-to-completion decision at this point. What we need: Keep and tBTC rewards to be visually consistent (2-3 panes, same width, ideally same height).

Let's do v2 from the designs, for both:

<img width="886" alt="Screen Shot 2020-11-30 at 09 40 53" src="https://user-images.githubusercontent.com/8245/100623435-2b01a500-32f0-11eb-85fe-e22cf88f57a0.png">

The beneficiary address can also be left off to get this shipped. For both rewards, let's also leave off the `?` next to the number of intervals left. For the beacon, we display both panes but skip the Withdraw All button.

Reply:
Regarding beneficiary address, if we display rewards from beneficiary's perspective, should be current logged account address, right?

Reply:
@Shadowfiend correct?
![obraz](https://user-images.githubusercontent.com/57687279/100754753-b90e8600-33eb-11eb-9872-e9909417906e.png)

Reply:
… Yeah, I think it makes less sense until we implement viewing from non-beneficiary addresses.